zerver/lib/user_groups.py

```diff
@@ -29,6 +29,24 @@ class UserGroupDict(TypedDict):
     can_mention_group_id: int
 
 
+def user_ids_to_users(user_ids: Sequence[int], realm: Realm) -> List[UserProfile]:
+    # TODO: Consider adding a flag to control whether deactivated
+    # users should be included.
+
+    user_profiles = list(UserProfile.objects.filter(id__in=user_ids).select_related("realm"))
```

Comment:
Do we need to select "realm" here now since we are directly accessing `realm_id` below? And the caller of this already has the realm object since it is passed into this function.

Reply:
Good point, I was thinking along the same lines, but I chickened out.

The problem is that the callers still end up passing the UserProfile objects down the stack, and I don't have the time to audit that whole call stack to make sure that nobody calls `user.realm` downstream.  Since these endpoints are called relatively rarely, I am not too concerned here about having super slim user objects, since they do a lot more real work like sending out messages to everybody added to a group, and an O(N)-fetching error there would be kinda embarrassing.